resources/fetching/fetchers/gcp/assets_fetcher.go

```diff
@@ -0,0 +1,118 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package fetchers
+
+import (
+	"context"
+
+	"cloud.google.com/go/asset/apiv1/assetpb"
+	"github.com/elastic/elastic-agent-libs/logp"
+
+	"github.com/elastic/cloudbeat/resources/fetching"
+	"github.com/elastic/cloudbeat/resources/providers/gcplib/inventory"
+)
+
+type GcpAssetsFetcher struct {
+	log        *logp.Logger
+	resourceCh chan fetching.ResourceInfo
+	provider   inventory.InventoryService
+}
+
+type GcpAsset struct {
+	Type    string
+	SubType string
+
```

Comment:
Are those fields really needed?
Before sending to eval we enrich the resource by calling the `GetMetadata` func.
https://github.com/elastic/cloudbeat/blob/b4431769070a7bd00f5959469c993daef6694b6a/evaluator/opa.go#L111-L120

Reply:
i think we do, because `GetMetadata` reads from those fields

Reply:
ok.
Then we can change GetData to return only the resource.
```
func (r *GcpAsset) GetData() interface{} {
	return r.Asset
}
```